triggers/daml/Daml/Trigger.daml

```diff
@@ -268,52 +268,59 @@ dedupExerciseByKey k c = do
   unless (any ((Some (k, c) ==) . fromExerciseByKey @t) cmds) $
     void $ emitCommands [exerciseByKeyCmd @t k c] []
 
--- | Transform the high-level trigger type into the one from `Daml.Trigger.LowLevel`.
-runTrigger : Trigger s -> LowLevel.Trigger (TriggerState s)
-runTrigger userTrigger = LowLevel.Trigger
-  { initialState = initialState
-  , update = update
-  , registeredTemplates = userTrigger.registeredTemplates
-  , heartbeat = userTrigger.heartbeat
-  }
-  where
-    initialState party readAs (ActiveContracts createdEvents) =
-      let acs = foldl (\acs created -> applyEvent (CreatedEvent created) acs) (ACS mempty Map.empty) createdEvents
-          userState = runTriggerInitializeA userTrigger.initialize (TriggerInitState acs party readAs)
-          state = TriggerState acs party readAs userState Map.empty
-      in TriggerSetup $ execStateT (runTriggerRule $ runRule userTrigger.rule) state
-    utUpdateState commandsInFlight acs actAs readAs msg =
+
+utUpdateState : (Message -> TriggerUpdateA s ()) -> Map CommandId [Command] -> ACS -> Party -> [Party] -> Message -> s -> s
+utUpdateState updateState commandsInFlight acs actAs readAs msg =
       let state = TriggerUpdateState commandsInFlight acs actAs readAs
-      in execState $ flip runTriggerUpdateA state $ userTrigger.updateState msg
-    update msg = do
-      time <- getTime
+      in execState $ flip runTriggerUpdateA state $ updateState msg
+
+updateImpl : Trigger s -> Bool -> [Message] -> TriggerRule (TriggerState s) ()
```

Comment:
I do not see how to transform that into a `foldl`. 
I would need something like `forA_` with an addition way to handle to accumulator `userTrigger` to prevent rule triggering.

Reply:
@akrmn has kindly helped out here.